x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations_task.ts

```diff
@@ -0,0 +1,221 @@
+    }
+  };
+
+  private get taskId(): string {
+    return `${TYPE}:${VERSION}`;
+  }
+
+  private endRun(msg: string = '') {
+    this.logger.info(`[SyncIntegrationsTask] runTask ended${msg ? ': ' + msg : ''}`);
+  }
+
+  public runTask = async (taskInstance: ConcreteTaskInstance, core: CoreSetup) => {
+    if (!this.wasStarted) {
+      this.logger.debug('[SyncIntegrationsTask] runTask Aborted. Task not started yet');
+      return;
+    }
+    // Check that this task is current
+    if (taskInstance.id !== this.taskId) {
+      this.logger.debug(
+        `[SyncIntegrationsTask] Outdated task version: Got [${taskInstance.id}] from task instance. Current version is [${this.taskId}]`
+      );
+      return getDeleteTaskRunResult();
+    }
+
+    this.logger.info(`[runTask()] started`);
+
+    const [coreStart] = await core.getStartServices();
+    const esClient = coreStart.elasticsearch.client.asInternalUser;
+    const soClient = new SavedObjectsClient(coreStart.savedObjects.createInternalRepository());
+
+    const { enableSyncIntegrationsOnRemote } = appContextService.getExperimentalFeatures();
+
+    if (!enableSyncIntegrationsOnRemote) {
+      return;
+    }
+
+    const indexExists = await esClient.indices.exists(
+      {
+        index: FLEET_SYNCED_INTEGRATIONS_INDEX_NAME,
+      },
+      { signal: this.abortController.signal }
+    );
+
+    if (!indexExists) {
+      this.logger.info(
+        `[SyncIntegrationsTask] index ${FLEET_SYNCED_INTEGRATIONS_INDEX_NAME} does not exist`
+      );
+      return;
+    }
+
+    try {
+      await this.updateSyncedIntegrationsData(esClient, soClient);
+
+      this.endRun('success');
+    } catch (err) {
+      if (err instanceof errors.RequestAbortedError) {
+        this.logger.warn(`[SyncIntegrationsTask] request aborted due to timeout: ${err}`);
+        this.endRun();
+        return;
+      }
+      this.logger.error(`[SyncIntegrationsTask] error: ${err}`);
+      this.endRun('error');
+    }
+  };
+
+  private updateSyncedIntegrationsData = async (
+    esClient: ElasticsearchClient,
+    soClient: SavedObjectsClient
+  ) => {
+    const outputs = await outputService.list(soClient);
+    const remoteESOutputs = outputs.items.filter(
+      (output) => output.type === outputType.RemoteElasticsearch
+    );
+    const isSyncEnabled = remoteESOutputs.some(
+      (output) => (output as NewRemoteElasticsearchOutput).sync_integrations
+    );
+
+    if (!isSyncEnabled) {
+      return;
+    }
+
+    const newDoc: SyncIntegrationsData = {
+      remote_es_hosts: remoteESOutputs.map((output) => {
+        const remoteOutput = output as NewRemoteElasticsearchOutput;
+        return {
+          name: remoteOutput.name,
+          hosts: remoteOutput.hosts ?? [],
+          sync_integrations: remoteOutput.sync_integrations ?? false,
+        };
+      }),
+      integrations: [],
+    };
+
+    const packageSavedObjects = await getInstalledPackageSavedObjects(soClient, {
```

Comment:
Given the "search limit" passed in, seems like this might be the "expensive" part of the task.  Perhaps the `abortController` could be passed in here as well, to get the task to stop "quicker" when it's cancelled.

Reply:
Ah, maybe not!  The main concern we typically have is if something is looping, without some reasonable bounds.  Looking at the implementation of `getInstalledPackageSavedObjects()`, it doesn't look like it's going to be a problem, so ... ignore my comment I guess :-)